### PR TITLE
Fix custom datadir (#771)

### LINKED ISSUE #771: kubelet fails to start if data dir on worker is different than on controller
**Version**
```
$ k0s version 0.11
```
**Platform**
Which platform did you run k0s on?
```
alpine linux 3.13, x86_64
```
**What happened?**
worker fails to start when data directory does not correspond with `--data-dir` on controller

**How To Reproduce**

* Start controller as non-root with data dir where it has write permissions: `k0s controller --data-dir /tmp/k0s`
* create a join token and join a worker (as root) without setting `--data-dir`.

kubelet fails to start

**Expected behavior**
k0s worker should not care about the `--data-dir` setting on the controller

**Screenshots & Logs**
```...
INFO[2021-03-11 16:08:41] F0311 16:08:41.808708    2250 server.go:257] unable to load client CA file /tmp/k0s/pki/ca.crt: open /tmp/k0s/pki/ca.crt: no such file or directory  component=kubelet
INFO[2021-03-11 16:08:41] goroutine 1 [running]:                        component=kubelet
INFO[2021-03-11 16:08:41] k8s.io/kubernetes/vendor/k8s.io/klog/v2.stacks(0xc000010001, 0xc000181b80, 0x94, 0x9e)  component=kubelet
...
WARN[2021-03-11 16:08:41] exit status 255                               component=kubelet
INFO[2021-03-11 16:08:41] respawning in 5s                              component=kubelet
```

**Additional context**

### PR DESCRIPTION
**Pull request by ncopa**
---
name: Pull Request
about: Create a Pull Request
title: ''
labels: ''
assignees: ''

---

**Issue**
Fixes #771

**What this PR Includes**
- Adds tests to verify issue #771 
- Clean up Makefile a bit (fix a warning and respect GO env var)
- fix kubernetes 1.20 leftovers
- fix a filehandle leak spotted by @jnummelin
- save the kubelet config as a template in configmap, so we can replace the path on worker side
- use text/template instead of html/template in template writer so the special chars are not html-escaped

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -66,6 +66,7 @@ require (
 	k8s.io/kubectl v0.20.2
 	k8s.io/mount-utils v0.20.4
 	k8s.io/utils v0.0.0-20201110183641-67b214c5f920
+	k8s.io/kubelet v0.20.5
```
Comment (jnummelin) [inline go.mod:69]: I don't think we need this dependency now with the templating approach

File: pkg/component/worker/kubeletconfigclient.go [hunk 2]
```diff
@@ -55,3 +57,17 @@ func (k *KubeletConfigClient) Get(profile string) (string, error) {
 	}
 	return config, nil
 }
+
+// Get reads the config from kube api
+func (k *KubeletConfigClient) GetAsStruct(profile string) (*kubeletconfig.KubeletConfiguration, error) {
```
Comment (jnummelin) [inline pkg/component/worker/kubeletconfigclient.go:62]: This func is not really used

**Review by jnummelin — CHANGES_REQUESTED** [review]
Some "dead code" found. In general looks good though

**Comment by jnummelin** [comment]
@jasmingacic @soider PTAL also, I'm bit too biased to review as I helped to conceive the whole template-in-template approach... 😄

**Review by jasmingacic — DISMISSED** [review]
Looks legit

File: go.mod (hunk 1 above)
Comment (ncopa) [reply go.mod]: :+1: cleaned up

File: pkg/component/worker/kubeletconfigclient.go (hunk 2 above)
Comment (ncopa) [reply pkg/component/worker/kubeletconfigclient.go]: removed.

**Review by jasmingacic — APPROVED** [review]

**Review by jnummelin — APPROVED** [review]

**Pull request merged**